Source/JavaScriptCore/runtime/JSString.h

```diff
@@ -389,16 +414,17 @@ class JSRopeString final : public JSString {
 
         static constexpr ptrdiff_t offsetOfLength() { return OBJECT_OFFSETOF(CompactFibers, m_length); }
         static constexpr ptrdiff_t offsetOfFiber1() { return OBJECT_OFFSETOF(CompactFibers, m_length); }
-        static constexpr ptrdiff_t offsetOfFiber2() { return OBJECT_OFFSETOF(CompactFibers, m_fiber1Upper); }
+        static constexpr ptrdiff_t offsetOfFiber2() { return OBJECT_OFFSETOF(CompactFibers, m_latter.m_fiber1Upper); }
 
     private:
         friend class LLIntOffsetsExtractor;
 
         uint32_t m_length { 0 };
         uint32_t m_fiber1Lower { 0 };
-        uint16_t m_fiber1Upper { 0 };
-        uint16_t m_fiber2Lower { 0 };
-        uint32_t m_fiber2Upper { 0 };
+        union {
+            Latter m_latter { };
+            uint64_t m_latterBits;
```

Comment:
I would make this an `Atomic<uint64_t>` just to make it clear you're setting this concurrently, Clang always compiles a relaxed store to a regular store anyway.